ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StepSchemaResolver.java

```diff
@@ -169,13 +172,29 @@ private LogicalSchema handleStreamSelect(
     ).getSchema();
   }
 
+  private LogicalSchema handleSelectKey(
+      final LogicalSchema sourceSchema,
+      final StreamSelectKey step
+  ) {
+    final ExpressionTypeManager expressionTypeManager =
```

Comment:
Can we move this out of this class? Ideally this class should just be routing to something else that owns the schema transformation. We can move it to the step builder for now.